src/app/meds-n-labels/activity-1/matchmeds.component.ts

```diff
@@ -0,0 +1,297 @@
+import { Component, OnInit, AfterViewInit, ViewChild, HostListener, ViewContainerRef } from '@angular/core';
+import { Coordinates } from './coordinate-structure';
+import { MatchingInfo } from './matching.info';
+import { ToastsManager } from 'ng2-toastr/ng2-toastr';
+
+@Component({
+  selector: 'app-matchmeds',
+  templateUrl: './matchmeds.component.html',
+  styleUrls: ['./matchmeds.component.scss']
+})
+
+
+export class MatchmedsComponent implements OnInit {
+    public position = 'col-md-10 col-md-offset-2';
+    // HTML Element for DOM Manipulations
+    @ViewChild('canv') canv;
+    canvasElement: HTMLCanvasElement;
+    cx: CanvasRenderingContext2D;
+    canvasTop: number;
+    canvasLeft: number;
+    height = 400;
+    width = 75;
+    windowWidth = window.innerWidth;
+    /**
+     * Default 75
+     * >400 - 100
+     * >768 - 200
+     * >992 - 250
+     */
+    // Number of Elements for Matching
+    numElements = MatchingInfo.numElements;
+    medicines = MatchingInfo.medicines;
+
+    display = MatchingInfo.medicinesideeffects;
+    // Counter for Keeping track of Element number
+    count = 0 ;
+    matchingComplete = 0;
+    // Start and End Points of a Line
+    start: Coordinates[] = [];
+    end: Coordinates[] = [];
+    // Temporaty Start and End Points to Handle Incomplete Drwaings
+    tempStart: Coordinates = {x: 0, y: 0};
+    tempEnd: Coordinates = {x: 0, y: 0};
+    // A flag to check if mouse is clicked through out the drawing
+    clicked = false;
+
+    // Canvas Position Information
+    radius = 10;
+    lineWidth = 5;
+    spacing = this.radius;
+    gap = 7 * (this.radius + this.spacing);
+
+    startPos = 0 + this.lineWidth;
```

Comment:
0 represents the actual starting position but since there is some line width to the circles we have to add them as offset.  I can change it but I thought this way it was more easily understood and for changing the look if anyone wanted to add some offset to it, the 0 can be changed to that value.  Should I change it to that ?